### ISSUE TITLE
avg_slopes ignoring weights

### ISSUE THREAD

**Issue by trose64** [vincentarelbundock/marginaleffects#961]
When calling avg_slopes with wts specified in the example below it appears to ignore the weights. Using avg_comparisons however gives the expected results where the weights are applied correctly. Not sure if this is a bug or the expected behavior. 

```r 

library(marginaleffects)

dat <- mtcars

set.seed(2354)

#make up weights and factor vars
dat$fake_wt <-runif(nrow(dat)) 
dat$hi_mpg <- as.factor(as.numeric(dat$mpg > 19))
dat$cyl <- as.factor(as.numeric(dat$cyl==6))
dat$gear <- as.factor(dat$gear)

mod <- glm(hi_mpg ~ cyl + gear + wt, data = dat, weights=dat$fake_wt, family="binomial")

#with weights avg_slopes and avg_comparison results differ (avg_slopes gives same results as without the wts)
avg_slopes(mod, variables = "cyl", by="gear", wts="fake_wt", type="response") 
avg_comparisons(mod, variables="cyl",  by="gear", wts="fake_wt", type="response") 

sessionInfo()

```

```
> library(marginaleffects)

> dat <- mtcars

> set.seed(2354)

> #make up weights and factor vars
> dat$fake_wt <-runif(nrow(dat)) 

> dat$hi_mpg <- as.factor(as.numeric(dat$mpg > 19))

> dat$cyl <- as.factor(as.numeric(dat$cyl==6))

> dat$gear <- as.factor(dat$gear)

> mod <- glm(hi_mpg ~ cyl + gear + wt, data = dat, weights=dat$fake_wt, family="binomial")

> #with weights avg_slopes and avg_comparison results differ (avg_slopes gives same results as without the wts)
> avg_slopes(mod, variables = "cyl", b .... [TRUNCATED] 

 Term          Contrast gear Estimate Std. Error      z Pr(>|z|)   S  2.5 % 97.5 %
  cyl mean(1) - mean(0)    3  -0.0330      0.184 -0.180    0.857 0.2 -0.393  0.327
  cyl mean(1) - mean(0)    4  -0.0206      0.117 -0.176    0.860 0.2 -0.250  0.208
  cyl mean(1) - mean(0)    5  -0.0339      0.194 -0.174    0.862 0.2 -0.415  0.347

Columns: term, contrast, gear, estimate, std.error, statistic, p.value, s.value, conf.low, conf.high, predicted_lo, predicted_hi, predicted 
Type:  response 


> avg_comparisons(mod, variables="cyl",  by="gear", wts="fake_wt", type="response") 

 Term          Contrast gear Estimate Std. Error      z Pr(>|z|)   S  2.5 % 97.5 %
  cyl mean(1) - mean(0)    3  -0.0288      0.158 -0.182    0.856 0.2 -0.339  0.282
  cyl mean(1) - mean(0)    4  -0.0243      0.138 -0.176    0.861 0.2 -0.296  0.247
  cyl mean(1) - mean(0)    5  -0.0389      0.224 -0.174    0.862 0.2 -0.478  0.400

Columns: term, contrast, gear, estimate, std.error, statistic, p.value, s.value, conf.low, conf.high, predicted_lo, predicted_hi, predicted 
Type:  response 


> sessionInfo()
R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8    LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                           LC_TIME=English_United States.utf8    

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] marginaleffects_0.16.0

loaded via a namespace (and not attached):
 [1] compiler_4.3.0    backports_1.4.1   cli_3.6.1         generics_0.1.3    tools_4.3.0       rstudioapi_0.14  
 [7] Rcpp_1.0.10       data.table_1.14.8 insight_0.19.1    checkmate_2.2.0   rlang_1.1.1 

```

```

**Comment by vincentarelbundock** [comment]
Thanks a lot for the report\! This should be fixed in version 0.16.0.9016 on github.

``` r
library(marginaleffects)
set.seed(2354)

#make up weights and factor vars
dat <- mtcars
dat$fake_wt <-runif(nrow(dat)) 
dat$hi_mpg <- as.factor(as.numeric(dat$mpg > 19))
dat$cyl <- as.factor(as.numeric(dat$cyl==6))
dat$gear <- as.factor(dat$gear)

mod <- glm(hi_mpg ~ cyl + gear + wt, data = dat, weights=dat$fake_wt, family="binomial")
# Warning in eval(family$initialize): non-integer #successes in a binomial glm!

s <- slopes(mod, variables = "cyl", type = "response", newdata = dat)

weighted.mean(s[s$gear == 3, "estimate"], s[s$gear == 3, "fake_wt"])
# [1] -0.02881515
mean(s[s$gear == 3, "estimate"])
# [1] -0.03302415

avg_slopes(mod, variables = "cyl", by="gear", wts="fake_wt", type="response")[1, "estimate"]
# [1] -0.02881515
```

**Comment by trose64** [comment]
Thanks @vincentarelbundock for the quick response! Is 0.16.0.901**6** an upcoming version? When I try to install the latest version from Github I'm only seeing 0.16.0.901**5**.
